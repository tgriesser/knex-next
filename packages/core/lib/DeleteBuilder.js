"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const WhereClauseBuilder_1 = require("./clauses/WhereClauseBuilder");
const datatypes_1 = require("./data/datatypes");
const SelectBuilder_1 = require("./SelectBuilder");
class DeleteBuilder extends WhereClauseBuilder_1.WhereClauseBuilder {
    constructor(ast = datatypes_1.deleteAst) {
        super();
        this.ast = ast;
    }
    from(tableName) {
        return this.chain(ast => ast.set("table", tableName));
    }
    getAst() {
        return this.ast;
    }
    subQuery(fn) {
        const builder = new SelectBuilder_1.SelectBuilder();
        fn.call(builder, builder);
        return new datatypes_1.SubQueryNode({ ast: builder.getAst() });
    }
    toOperation() {
        return this.grammar.toOperation(this.ast);
    }
    chain(fn) {
        this.ast = fn(this.ast);
        return this;
    }
}
exports.DeleteBuilder = DeleteBuilder;
