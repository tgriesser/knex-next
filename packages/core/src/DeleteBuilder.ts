import { WhereClauseBuilder } from "./clauses/WhereClauseBuilder";
import { ChainFnDelete, SubQueryArg } from "./data/types";
import { deleteAst, SubQueryNode } from "./data/datatypes";
import { SelectBuilder } from "./SelectBuilder";

export class DeleteBuilder extends WhereClauseBuilder {
  constructor(protected ast = deleteAst) {
    super();
  }

  from(tableName: string) {
    return this.chain(ast => ast.set("table", tableName));
  }

  getAst() {
    return this.ast;
  }

  subQuery(fn: SubQueryArg) {
    const builder = new SelectBuilder();
    fn.call(builder, builder);
    return new SubQueryNode({ ast: builder.getAst() });
  }

  toOperation() {
    return this.grammar.toOperation(this.ast);
  }

  protected chain(fn: ChainFnDelete) {
    this.ast = fn(this.ast);
    return this;
  }
}
