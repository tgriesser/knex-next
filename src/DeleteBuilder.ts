import { WhereClauseBuilder } from "./WhereClauseBuilder";
import { ChainFnDelete } from "./types";
import { deleteAst } from "./datatypes";

export class DeleteBuilder extends WhereClauseBuilder {
  constructor(protected ast = deleteAst) {
    super();
  }
  getAst() {
    return this.ast;
  }
  protected chain(fn: ChainFnDelete) {
    this.ast = fn(this.ast);
    return this;
  }
}
