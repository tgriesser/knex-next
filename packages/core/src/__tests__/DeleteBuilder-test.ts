import { DeleteBuilder } from "../DeleteBuilder";
import { commonDeleteTests } from "../testing/common-delete-tests";

describe("@knex/core - DeleteBuilder", () => {
  commonDeleteTests(() => new DeleteBuilder());
});
