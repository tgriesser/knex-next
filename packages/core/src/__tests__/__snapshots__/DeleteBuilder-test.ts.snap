// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@knex/core - DeleteBuilder from table 1`] = `
Immutable.DeleteOperation {
  "__operation": "DELETE",
  "table": "someTable",
  "where": Immutable.List [],
}
`;

exports[`@knex/core - DeleteBuilder from table 2`] = `
Object {
  "fragments": Array [
    "DELETE FROM someTable",
  ],
  "query": "DELETE FROM someTable",
  "sql": "DELETE FROM someTable",
  "values": Array [],
}
`;

exports[`@knex/core - DeleteBuilder from table where 1`] = `
Immutable.DeleteOperation {
  "__operation": "DELETE",
  "table": "someTable",
  "where": Immutable.List [
    Immutable.WhereExpressionNode {
      "__typename": "WhereExpressionNode",
      "not": null,
      "column": "id",
      "operator": "=",
      "value": 1,
      "andOr": "AND",
    },
  ],
}
`;

exports[`@knex/core - DeleteBuilder from table where 2`] = `
Object {
  "fragments": Array [
    "DELETE FROM someTable WHERE id = ",
    "",
  ],
  "query": "DELETE FROM someTable WHERE id = ?",
  "sql": "DELETE FROM someTable WHERE id = 1",
  "values": Array [
    1,
  ],
}
`;
